[flagd-provider] package-private interface prevents resolverType option usage with Kotlin

The [EvaluatorType](https://github.com/open-feature/java-sdk-contrib/blob/60b14465389d01c9ef1d7d0323a325cd744415a3/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/Config.java#L98) in flagd-providers `Config.java` is package private.

A bug in Kotlin prevents the usage of the public enum `Config.Resolver` as it implements the package private interface: https://youtrack.jetbrains.com/issue/KT-11700

Due to this bug the code 

```
    fun getFlagdProvider(): FlagdProvider = FlagdProvider(
        FlagdOptions.builder()
            .resolverType(Config.Resolver.IN_PROCESS)
            .build()
    )
```

throws an IllegalAccessError at runtime with the cause

```
Caused by: java.lang.IllegalAccessError: failed to access class dev.openfeature.contrib.providers.flagd.Config$EvaluatorType from class x.y.z.MyClass (dev.openfeature.contrib.providers.flagd.Config$EvaluatorType and x.y.z.MyClass are in unnamed module of loader 'app')
```

--- a/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/Config.java
+++ b/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/Config.java
@@ -94,8 +94,10 @@ public final class Config {
         }
     }
 
-    // intermediate interface to unify deprecated Evaluator & new Resolver
-    interface EvaluatorType {
+    /**
+     * intermediate interface to unify deprecated Evaluator and new Resolver.
+     **/
+    public interface EvaluatorType {
         String asString();
     }
 

